official/keras_application_models/model_callbacks.py

```diff
@@ -0,0 +1,159 @@
+
+_METRICS_TO_LOG = {
+    "loss": "train_loss",
+    "acc": "train_accuracy",
+    "val_loss": "loss",
+    "val_acc": "accuracy"
+}
+
+
+class ExamplesPerSecondCallback(tf.keras.callbacks.Callback):
+  """ExamplesPerSecond callback.
+
+  This callback records the average examples per second during training.
+  """
+
+  def __init__(self, batch_size=None, epoch_size=None, batch_based=None,
+               epoch_based=None, metric_logger=None):
+    if (batch_based is None) == (epoch_based is None):
+      raise ValueError("Exactly one of batch_based "
+                       "and epoch_based should be provided.")
+
+    if batch_based and (batch_size is None):
+      raise ValueError("batch_size should be provided for "
+                       "batch_based benchmark.")
+
+    if epoch_based and (epoch_size is None):
+      raise ValueError("epoch_size should be provided for "
+                       "epoch_based benchmark.")
+
+    self._batch_based = batch_based
+    self._epoch_based = epoch_based
+    self._batch_size = batch_size
+    self._epoch_size = epoch_size
+
+    self._logger = metric_logger or logger.BaseBenchmarkLogger()
+
+  def on_train_begin(self, logs=None):
+    self._global_step = 0
+    # For batch_based
+    if self._batch_based:
+      self._train_time_batch_based = 0
+    else:  # For epoch_based
+      self._epochs = 0
+      self._train_time_epoch_based = 0
+
+  def on_batch_begin(self, batch, logs=None):
+    if self._batch_based:
+      self._time_start_batch_based = time.time()
+
+  def on_batch_end(self, batch, logs=None):
+    self._global_step += 1
+    if self._batch_based:
+      self._train_time_batch_based += time.time() - self._time_start_batch_based
+      examples_per_sec_batch_based = self._batch_size * (
+          self._global_step / self._train_time_batch_based)
+      self._logger.log_metric(
+          "examples_per_sec_batch_based",
+          examples_per_sec_batch_based,
+          global_step=self._global_step)
+
+  def on_epoch_begin(self, epoch, logs=None):
+    if self._epoch_based:
+      self._time_start_epoch_based = time.time()
+
+  def on_epoch_end(self, epoch, logs=None):
+    if self._epoch_based:
+      self._epochs += 1
+      self._train_time_epoch_based += time.time() - self._time_start_epoch_based
+      examples_per_sec_epoch_based = self._epoch_size * (
+          self._epochs / self._train_time_epoch_based)
+      self._logger.log_metric(
+          "examples_per_sec_epoch_based",
+          examples_per_sec_epoch_based,
+          global_step=self._global_step)
+
+
+class LoggingMetricCallback(tf.keras.callbacks.Callback):
+  """LoggingMetric callback.
+
+  By default, four metrics are logged: train accuracy, train loss, evaluation
+  accuracy and evaluation loss. Check _METRICS_TO_LOG for details.
+  """
+
+  def __init__(self, metrics=None, batch_based=None, epoch_based=None,
+               metric_logger=None):
+    if (batch_based is None) == (epoch_based is None):
```

Comment:
This assumes these are None and not False, which seems fragile. Maybe if batch_based == epoch_based?